cmake/Modules/FindHWLOC.cmake

```diff
@@ -68,25 +65,26 @@ if (HWLOC_INCLUDE_DIRS)
         get_dec_from_hex("${HEX_PATCH}" DEC_PATCH)
         set(HWLOC_VERSION "${DEC_MAJOR}.${DEC_MINOR}.${DEC_PATCH}" CACHE STRING "HWLOC version")
     endif()
+endif()
 
-    if (NOT HWLOC_FIND_QUIETLY)
-        if (HWLOC_FOUND AND HWLOC_LIBRARIES)
-            message(STATUS "Looking for HWLOC - found version ${HWLOC_VERSION}")
-        else()
-            message(STATUS "${Magenta}Looking for HWLOC - not found"
-                "\n   Please check that your environment variable HWLOC_DIR"
-                "\n   has been set properly.${ColourReset}")
-        endif()
+if (NOT HWLOC_FIND_QUIETLY)
+    if (HWLOC_INCLUDE_DIRS AND HWLOC_LIBRARIES)
+        message(STATUS "Looking for HWLOC - found version ${HWLOC_VERSION}")
+    else()
+        message(STATUS "${Magenta}Looking for HWLOC - not found"
+            "\n   Please check that your environment variable HWLOC_DIR"
+            "\n   has been set properly.${ColourReset}")
+        unset(HWLOC_LIBRARIES CACHE)
+        unset(HWLOC_INCLUDE_DIRS CACHE)
```

Comment:
I could imagine that this might cause issues when users are trying to debug their FindHWLOC via CMakeCache.txt, when only one of the two paths is found correctly.

Reply:
Yes, but on the other hand if these are not cleaned then Ginkgo will build/install hwloc, but there is a risk that it will consider the headers or library from the installed version which wasn't complete, and the other one will come from the hwloc that we built... I've seen that happening quite a few times, which is why I decided to make sure the finder doesn't leave things in a weird state.